src/app/components/game-session/room/room.component.ts

```diff
@@ -1,14 +1,112 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ConversationModel } from '../../../models/conversation.model';
+import { ProfilService } from '../../../services/profil/profil.service';
+import { SocketService } from '../../../modules/call/services/socket.service';
+import { MessageToBackModel } from '../../../models/messageToBack.model';
+import { User } from '../../../models/user.model';
+import { AuthService } from '../../../services/auth/auth.service';
+import { ChatComponent } from '../../../modules/chat/components/chat/chat.component';
+import { JoinGameSessionChatDTO } from './dto/JoinGameSessionChatDTO';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { JoinGameSessionVisioDTO } from './dto/JoinGameSessionVisioDTO';
 
 @Component({
     selector: 'app-room',
     templateUrl: './room.component.html',
     styleUrls: ['./room.component.css'],
 })
 export class RoomComponent implements OnInit {
-    constructor() {}
+    public conversation: ConversationModel | undefined;
+    public me: User | null | undefined;
+    private gameSessionId: string | undefined;
+
+    @ViewChild(ChatComponent) chatComponent: ChatComponent | undefined;
+
+    constructor(
+        private profilService: ProfilService,
+        public socketService: SocketService,
+        private authService: AuthService,
+        private _snackBar: MatSnackBar,
+    ) {}
 
     isHideChat = true;
 
-    ngOnInit(): void {}
+    async ngOnInit(): Promise<void> {
+        this.me = this.authService.user;
+        const currentURL = window.location.href;
+        const parts = currentURL.split('/');
+        const gameSessionId = parts[parts.length - 1];
```

Comment:
utiliser ca plutot https://angular.io/tutorial/tour-of-heroes/toh-pt5#extract-the-id-route-parameter